feat(nuxt): Use --import as the default installation method (#12070)
getsentry/sentry-docs: docs/platforms/javascript/guides/nuxt/install/top-level-import.mdx

---
title: Top-level import
sidebar_order: 2
description: "Learn about how the Nuxt SDK adds a top-level import to the build output."
---

## Understanding top-level `import`

Sentry needs to be initialized before the application starts.
If the default way of adding an <PlatformLink to="/install/cli-import">`--import` CLI flag</PlatformLink> flag does not work for you,

> set up the SDK for adding a top-level `import`. This will import the Sentry server-side config at the top of the server entry file.

Reviewer: l: maybe being a bit more specific here is less confusing?

```suggestion
set up the SDK for adding a top-level `import`. This will import the Sentry server-side config at the top of the Nuxt server entry file.
```

> Sentry needs to be initialized before the application starts.
> If the default way of adding an <PlatformLink to="/install/cli-import">`--import` CLI flag</PlatformLink> flag does not work for you,
> set up the SDK for adding a top-level `import`. This will import the Sentry server-side config at the top of the server entry file.

Reviewer 2: ```suggestion
Sentry needs to be initialized before the application starts. If adding  the <PlatformLink to="/install/cli-import">`--import` CLI flag</PlatformLink> doesn’t work for you, add a top-level `import` in the server entry file to load Sentry’s server-side config.
```

> In this scenario, Sentry is not initialized before application startup, but as early as otherwise possible.

Reviewer 2: ```suggestion
In this case, Sentry isn’t initialized before the app starts, but is set up as early as possible.
```

<Alert level='warning' title='Restrictions of this installation method'>

>   This installation method has a fundamental restriction: It only supports limited performance instrumentation. Only basic `http` instrumentation will work, and no DB or framework-specific instrumentation will be available.

Reviewer: ```suggestion
  This installation method has fundamental restrictions: It only supports limited performance instrumentation. Only basic `http` instrumentation will work, and no DB or framework-specific instrumentation will be available.
```

>   Because of this, we recommend using this only if the `--import` flag is not an option for you.

Reviewer 2: ```suggestion
  We recommend using this only if the `--import` flag is not an option for you.
```

</Alert>

> ## Initializing Sentry with a top-level `import`

Reviewer 2: ```suggestion
## Initializing Sentry With a Top-level `import`
```

Enable the top-level `import` by setting `autoInjectServerSentry`:

```typescript {filename:nuxt.config.ts} {3}
export default defineNuxtConfig({
  sentry: {
    autoInjectServerSentry: 'top-level-import'
  },
})
```

> By default, the SDK will add the Sentry server config to the build output (typically `.output/server/sentry.server.config.mjs`).

Reviewer 2: ```suggestion
By default, the SDK will add the Sentry server config to the build output (typically, `.output/server/sentry.server.config.mjs`).
```

> With the top-level `import`, the server entry file will look something like this:

Reviewer: ```suggestion
With the top-level `import`, the Nuxt server entry file will look something like this:
```

```javascript {filename:.output/server/index.mjs}
import './sentry.server.config.mjs';

> // Note: The file may have some imports and code related to debug IDs

Reviewer 2: ```suggestion
// Note: The file may have some imports and code, related to debug IDs
```
